bash.sh

```diff
@@ -61,6 +104,34 @@ array_join() {
     IFS=$sep eval 'echo "$*"'
 }
 
+#
+# $1: name of the array variable
+# $2: index of element to remove
+#
+array_pop() {
```

Comment:
nit: `pop` makes me think about removing a single element: I was expecting a single element to be returned (first? last?)

what about `filter_by_index`? 

Also, random thought, you could re-use your `array_filter`, right? Could be a nice example on how to pass in functions in to `array_filter`.